src/ecalc/libraries/libecalc/common/libecalc/expression/expression_evaluator.py

```diff
@@ -325,6 +325,7 @@ def lexer(expression: Union[str, int, float]) -> List[Token]:
         (r"while", TokenTag.operator),
         (r"do", TokenTag.operator),
         (r"end", TokenTag.operator),
+        (r"[0-9]\.[0-9]+e-?[0-9]+", TokenTag.numeric),  # Scientific notation, e.g. 1.23e-05 or 3.4e7
```

Comment:
what about 1e-8? It would. Can this happen to big numbers as well? In that case 3e29 and 3e+29?